scripts/keras_classification.py

```diff
@@ -0,0 +1,738 @@
+            required=True,
+            help="The type of model to use, ex: resnet50, vgg16, mobilenet "
+            "put as help to see the full list (will raise an exception with the list)",
+        )
+        par.add_argument(
+            "--pretrained",
+            type=str,
+            default=True,
+            help="The type of pretrained weights to use, "
+            "default is true to load the default pretrained weights for the model. "
+            "Otherwise should be set to the desired weights type: "
+            "[base, optim, optim-perf]. "
+            "To not load any weights set to one of [none, false]",
+        )
+        par.add_argument(
+            "--pretrained-dataset",
+            type=str,
+            default=None,
+            help="The dataset to load pretrained weights for if pretrained is set. "
+            "Default is None which will load the default dataset for the architecture."
+            " Ex can be set to imagenet, cifar10, etc",
+        )
+        par.add_argument(
+            "--checkpoint-path",
+            type=str,
+            default=None,
+            help="A path to a previous checkpoint to load the state from and "
+            "resume the state for. If provided, pretrained will be ignored",
+        )
+        par.add_argument(
+            "--model-kwargs",
+            type=json.loads,
+            default={},
+            help="kew word arguments to be passed to model constructor, should be "
+            " given as a json object",
+        )
+
+        # dataset args
+        par.add_argument(
+            "--dataset",
+            type=str,
+            required=True,
+            help="The dataset to use for training, "
+            "ex: imagenet, imagenette, cifar10, etc. "
+            "Set to imagefolder for a generic dataset setup "
+            "with an image folder structure setup like imagenet or loadable by a "
+            "dataset in sparseml.keras.datasets",
+        )
+        par.add_argument(
+            "--dataset-path",
+            type=str,
+            required=True,
+            help="The root path to where the dataset is stored",
+        )
+        par.add_argument(
+            "--dataset-kwargs",
+            type=json.loads,
+            default={},
+            help="kew word arguments to be passed to dataset constructor, should be "
+            " given as a json object",
+        )
+
+        # logging and saving
+        par.add_argument(
+            "--model-tag",
+            type=str,
+            default=None,
+            help="A tag to use for the model for saving results under save-dir, "
+            "defaults to the model arch and dataset used",
+        )
+        par.add_argument(
+            "--save-dir",
+            type=str,
+            default="keras_classification",
+            help="The path to the directory for saving results",
+        )
+
+        # task specific arguments
+        if par == train_parser:
```

Comment:
why not add these outside of the for loop if they're only being applied to one?

Reply:
fixed